### PR TITLE
Cleanup mapper

### PR DESCRIPTION
**Pull request by skosito**
# Description

Fix mapping issues and prisma schema cases across the app.

## Related Issue

https://github.com/eea-oasis/baseline/issues/580
https://github.com/eea-oasis/baseline/issues/584

## Motivation and Context

Check issues for more details.

## How Has This Been Tested

Did regression testing of all postman requests.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/src/bri/identity/bpiAccounts/accounts.profile.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export class AccountsProfile extends AutomapperProfile {
   override get profile() {
     return (mapper) => {
       createMap(mapper, BpiAccount, BpiAccountDto);
+      createMap(mapper, BpiAccount, BpiAccount);
```
Comment (Manik-Jain) [inline examples/bri-3/src/bri/identity/bpiAccounts/accounts.profile.ts:16]: I think we don't need this

File: examples/bri-3/src/bri/workgroup/workflows/workflow.profile.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ export class WorkflowProfile extends AutomapperProfile {
   override get profile() {
     return (mapper) => {
       createMap(mapper, Workflow, WorkflowDto);
+      createMap(mapper, Workflow, Workflow);
```
Comment (Manik-Jain) [inline examples/bri-3/src/bri/workgroup/workflows/workflow.profile.ts:16]: I think we don't need this

File: examples/bri-3/src/bri/workgroup/workgroups/workgroups.profile.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+@Injectable()
+export class WorkgroupProfile extends AutomapperProfile {
+  constructor(@InjectMapper() mapper: Mapper) {
+    super(mapper);
+  }
+
+  override get profile() {
+    return (mapper) => {
+      createMap(mapper, Workgroup, Workgroup);
```
Comment (Manik-Jain) [inline examples/bri-3/src/bri/workgroup/workgroups/workgroups.profile.ts:15]: I think we don't need this

File: examples/bri-3/src/bri/workgroup/worksteps/workstep.profile.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ export class WorkstepProfile extends AutomapperProfile {
   override get profile() {
     return (mapper) => {
       createMap(mapper, Workstep, WorkstepDto);
+      createMap(mapper, Workstep, Workstep);
```
Comment (Manik-Jain) [inline examples/bri-3/src/bri/workgroup/worksteps/workstep.profile.ts:16]: I think we don't need this

File: examples/bri-3/src/bri/identity/bpiAccounts/accounts.profile.ts (hunk 1 above)
Comment (skosito) [reply examples/bri-3/src/bri/identity/bpiAccounts/accounts.profile.ts:16]: we need it, without it error that mapping between BpiAccount and BpiAccount doesn't exist is thrown

**Review by Manik-Jain — APPROVED** [review]

**Review by ognjenkurtic — APPROVED** [review]

**Pull request merged**